### PR TITLE
Mining mechas refactors

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

mostly equipment but some mining messes too
## Why It's Good For The Game

remove soul

## Changelog
:cl:
refactor: Mining mechas refactors
/:cl:

### CODE REVIEW

**Comment by zxaber** [comment]
Built-in ore boxes are by design a feature of Clarke mechs, not all mechs. Is that an intentional change here? Is there any reason for it?

**Comment by TiviPlus** [comment]
They dont have builtin boxes, it's just to track the active orebox instead of using locate() every time the mech moves

File: code/modules/vehicles/mecha/working/working.dm [hunk 1]
```diff
@@ -1,5 +1,15 @@
 /obj/vehicle/sealed/mecha/working
 	internal_damage_threshold = 60
+	/// Handles an internal ore box for working mechs
+	var/obj/structure/ore_box/box
+
+/obj/vehicle/sealed/mecha/working/Initialize()
+	. = ..()
+	box = new /obj/structure/ore_box(src)
```
Comment (Fikou) [inline code/modules/vehicles/mecha/working/working.dm:8]: ripley isnt supposed to have an internal ore box though, just clarke

File: code/modules/vehicles/mecha/equipment/tools/work_tools.dm [hunk 2]
```diff
@@ -11,24 +11,27 @@
...
+	var/obj/vehicle/sealed/mecha/working/ripley/cargo_holder
 
-/obj/item/mecha_parts/mecha_equipment/hydraulic_clamp/can_attach(obj/vehicle/sealed/mecha/working/ripley/M)
-	if(..())
-		if(istype(M))
-			return 1
-	return 0
+/obj/item/mecha_parts/mecha_equipment/hydraulic_clamp/can_attach(obj/vehicle/sealed/mecha/M)
+	if(!.)
+		return
```
Comment (Rohesie) [inline code/modules/vehicles/mecha/equipment/tools/work_tools.dm:25]:
```suggestion
/obj/item/mecha_parts/mecha_equipment/hydraulic_clamp/can_attach(obj/vehicle/sealed/mecha/M)
	. = ..()
```
Forgot to call parent here, also no need to check the parent return if I'm getting this logic correctly?

**Comment by Rohesie** [comment]
`snalke_case` > `gluedwordsmashup`

File: code/modules/vehicles/mecha/equipment/tools/work_tools.dm (hunk 2 above)
Comment (TiviPlus) [reply code/modules/vehicles/mecha/equipment/tools/work_tools.dm]: This needs the parent call and if(!.), added

**Comment by Rohesie** [comment]
Ready and tested?
Also, does this have any gameplay change? If not, the changelog is optional, but if it does, can you clarify it?

**Comment by TiviPlus** [comment]
Yes this is tested and should have no gameplay change

**Pull request merged**